sources/zh.godamanga/src/html/mod.rs

```diff
@@ -0,0 +1,202 @@
+				let author_names: Vec<String> = authors
+					.iter()
+					.filter_map(|author| author.as_object())
+					.filter_map(|author_obj| author_obj.get("name"))
+					.filter_map(|name| name.as_str())
+					.map(|s| s.to_string())
+					.collect();
+				if !author_names.is_empty() {
+					manga.authors = Some(author_names);
+				}
+			}
+
+			// Extract genres/tags
+			if let Some(genres) = obj.get("genre").and_then(|v| v.as_array()) {
+				let tags: Vec<String> = genres
+					.iter()
+					.filter_map(|genre| genre.as_str())
+					.map(|s| s.to_string())
+					.collect();
+				if !tags.is_empty() {
+					manga.tags = Some(tags);
+				}
+			}
+
+			// Extract status
+			if let Some(status) = obj.get("creativeWorkStatus").and_then(|v| v.as_str()) {
+				manga.status = match status {
+					"連載中" => MangaStatus::Ongoing,
+					"完結" | "已完結" => MangaStatus::Completed,
+					_ => MangaStatus::Unknown,
+				};
+			}
+
+			// Get mid from HTML (still needed for API calls)
+			let mid = html
+				.select_first("#mangachapters")
+				.and_then(|e| e.attr("data-mid"))
+				.unwrap_or_else(String::new);
+			manga.key = format!("{}/{}", ids[0], mid);
+			manga.url = Some(url);
+			return Ok(());
+		}
+
+		// Fallback to HTML parsing if JSON-LD parsing fails
+		let mid = html
+			.select_first("#mangachapters")
+			.and_then(|e| e.attr("data-mid"))
+			.unwrap_or_else(String::new);
+		manga.cover = html
+			.select_first("meta[property='og:image']")
+			.and_then(|e| e.attr("content"))
+			.map(handle_cover_url);
+		manga.title = html
+			.select_first("title")
+			.and_then(|e| e.text())
+			.map(|t| t.replace("-G站漫畫", ""))
+			.unwrap_or_else(String::new);
+		let author = html
+			.select("a[href*=author]>span")
+			.map(|elements| {
+				elements
+					.filter_map(|a| a.text().map(|t| t.replace(",", "")))
+					.filter(|a| !a.is_empty())
+					.collect::<Vec<String>>()
+					.join(", ")
+			})
+			.unwrap_or_else(String::new);
+		let description = html
+			.select_first(".text-medium.my-unit-md")
+			.and_then(|e| e.text())
+			.unwrap_or_else(String::new);
+		let categories = html
+			.select(".py-1>a:not([href*=author])>span")
+			.map(|elements| {
+				elements
+					.filter_map(|a| {
+						a.text().map(|t| {
+							t.replace(",", "")
+								.replace("热门漫画", "")
+								.replace("#", "")
+								.replace("热门推荐", "")
+								.trim()
+								.to_string()
+						})
+					})
+					.filter(|a| !a.is_empty())
+					.collect::<Vec<String>>()
+			})
+			.unwrap_or_else(Vec::new);
+		let status = MangaStatus::Ongoing;
+
+		manga.key = format!("{}/{}", ids[0], mid);
+		manga.authors = Some(vec![author]);
+		manga.description = Some(description);
+		manga.tags = Some(categories);
+		manga.status = status;
+		manga.url = Some(url);
+
+		Ok(())
+	}
+
+	fn manga_page_result(&self) -> Result<MangaPageResult> {
+		let mut entries: Vec<Manga> = Vec::new();
+
+		for item in self.try_select(".pb-2>a")? {
+			let href = item.attr("href").unwrap_or_else(String::new);
```

Comment:
this should do the same thing but is more conventional
```suggestion
			let href = item.attr("href").unwrap_or_default();
```